propolis/src/exits.rs

```diff
@@ -1,13 +1,21 @@
+//! Describes transitions from VMs to the VMM.
+
 use std::convert::TryFrom;
 use std::os::raw::c_void;
 
 use bhyve_api::{
     vm_entry, vm_entry_cmds, vm_entry_payload, vm_exit, vm_exitcode,
 };
 
+/// Describes the reason for exiting execution of a vCPU.
 pub struct VmExit {
+    /// The instruction pointer of the guest at the time of exit.
     pub rip: u64,
+    /// The length of the instruction which triggered the exit.
+    /// Zero means unknown.
+    // TODO: Unsure about this; is this description accurate?
```

Comment:
For exits which are due to an instruction (say an attempted MMIO), it's the length of that instruction.  There are other exits reasons which are not related to the currently executing guest instruction which would result in that field being 0.